webhook_to_fedora_messaging_messages/forgejo/v1.py

```diff
@@ -0,0 +1,80 @@
+# SPDX-FileCopyrightText: 2024-2025 Contributors to the Fedora Project
+#
+# SPDX-License-Identifier: LGPL-3.0-or-later
+
+import typing
+from hashlib import sha1, sha256
+
+from ..base import Webhook2FedMsgBase
+from .utils import serialize_body, summarize_repository_event
+
+
+class ForgejoMessageV1(Webhook2FedMsgBase):
+    @property
+    def app_name(self) -> str:
+        """Application name on Fedora Messaging"""
+        return "Forgejo"
+
+    @property
+    def target_id(self) -> str:
+        """Webhook identifier specific to Forgejo"""
+        return self.body["body"]["repository"]["id"]
+
+    @property
+    def delivery(self) -> str:
+        """Unique identifier (GUID) for event"""
+        return self.body["headers"]["x-forgejo-delivery"]
+
+    @property
+    def signature(self) -> str:
+        """SHA160 signature of the request - Computed"""
+        return f"sha1={sha1(serialize_body(self.body).encode(encoding='utf-8')).hexdigest()}"  # noqa: S324
```

Comment:
The `.encode(encoding='utf-8')` is used in both places, it could be part of `serialize_body()`.